### PR TITLE
docs: Update public-facing packages' READMEs to include common template contents

### PR DESCRIPTION
**Pull request by Josmithr**
Also makes some minor organization updates for consistency.

### CODE REVIEW

File: packages/dds/counter/README.md [hunk 1]
```diff
@@ -15,7 +15,25 @@ library consumers should always prefer `^`.
...
+
+To access the `public` ([SemVer](https://semver.org/)) APIs, import via `@fluidframework/counter` like normal.
+
+To access the `legacy` APIs, import via `@fluidframework/counter/legacy`.
+
+<!-- prettier-ignore-end -->
+
+<!-- AUTO-GENERATED-CONTENT:END -->
+
+## SharedCounter
```
Comment (Josmithr) [inline packages/dds/counter/README.md:36]: Renamed heading to match the structure of `map`'s README.

File: packages/dds/counter/README.md [hunk 2]
```diff
@@ -153,24 +171,42 @@ This causes the `incremented` event to be sent to all of the clients who have th
 
 Since `updateCounterValueLabel` is listening for all `incremented` events, the view will always refresh with the appropriate updated value any time a collaborator increments or decrements the counter value.
 
-<!-- AUTO-GENERATED-CONTENT:START (README_API_DOCS_SECTION:includeHeading=TRUE) -->
+<!-- AUTO-GENERATED-CONTENT:START (LIBRARY_PACKAGE_README:scripts=FALSE&installation=FALSE&importInstructions=FALSE) -->
```
Comment (Josmithr) [inline packages/dds/counter/README.md:174]: Opted to use the "whole-readme" template (excluding the 2 sections placed towards the top of the README) to simplify things and to ensure future contents appear automatically as we add them.

Longer term, we may want to add package README "header" and "footer" pragmas to simplify this pattern, which seems to have become the standard across the repo.

File: packages/dds/counter/README.md [hunk 3]
```diff
@@ -153,24 +171,42 @@ This causes the `incremented` event to be sent to all of the clients who have th
...
+This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
+For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.
+
+This project may contain Microsoft trademarks or logos for Microsoft projects, products, or services.
+Use of these trademarks or logos must follow Microsoft’s [Trademark & Brand Guidelines](https://www.microsoft.com/trademarks).
+Use of Microsoft trademarks or logos in modified versions of this project must not cause confusion or imply Microsoft sponsorship.
+
+## Help
+
+Not finding what you're looking for in this README? Check out our [GitHub
```
Comment (Josmithr) [inline packages/dds/counter/README.md:203]: This probably wants to point to our website first, and github second (or not at all). I can make an update to this template in a separate follow-up.

Comment (Josmithr) [reply packages/dds/counter/README.md:203]: #21736

**Review by alexvy86 — APPROVED** [review]

File: packages/service-clients/odsp-client/README.md [hunk 4]
```diff
@@ -2,6 +2,39 @@
...
+<!-- AUTO-GENERATED-CONTENT:START (README_DEPENDENCY_GUIDELINES_SECTION:includeHeading=TRUE) -->
+
+<!-- prettier-ignore-start -->
+<!-- NOTE: This section is automatically generated using @fluid-tools/markdown-magic. Do not update these generated contents directly. -->
+
+## Using Fluid Framework libraries
+
+When taking a dependency on a Fluid Framework library, we recommend using a `^` (caret) version range, such as `^1.3.4`.
+While Fluid Framework libraries may use different ranges with interdependencies between other Fluid Framework libraries,
+library consumers should always prefer `^`.
```
Comment (jason-ha) [inline packages/service-clients/odsp-client/README.md:14]: This is only true is consuming public support level APIs include `fluid-framework/legacy`. For any consummation of `/beta` or `/alpha` APIs we recommend `~` dependencies.

File: packages/service-clients/odsp-client/README.md [hunk 5]
```diff
@@ -2,6 +2,39 @@
 
 The odsp-client package provides a simple and powerful way to consume collaborative Fluid data with OneDrive/SharePoint (ODSP) storage. Please note that odsp-client is currently an experimental package. We'd love for you to try it out and provide feedback but it is not yet recommended or supported for production scenarios.
```
Comment (jason-ha) [inline packages/service-clients/odsp-client/README.md:3]: Pre-existing, @ChumpChief + @sonalideshpandemsft , do we want to continue saying "experimental" as this time?

File: packages/service-clients/odsp-client/README.md (hunk 4 above)
Comment (Josmithr) [reply packages/service-clients/odsp-client/README.md:14]: Good callout. I'll put up a separate PR to update the wording of that particular template.

Comment (Josmithr) [reply packages/service-clients/odsp-client/README.md:14]: #21740

**Review by jason-ha — APPROVED** [review]

**Pull request merged**